src/Database/Type/StringType.php

```diff
@@ -94,4 +94,14 @@ public function marshal($value)
         }
         return (string)$value;
     }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return boolean False as databse results are returned already as strings
+     */
+    public function requiresToPHPCast()
```

Comment:
I think similar to https://github.com/cakephp/cakephp/blob/master/src/Database/Type/BoolType.php#L57 this should be called

    requiresToPhpCast()

(lowercase hp)